### ISSUE TITLE
window.parent should have the same global variables as window

### ISSUE THREAD

**Issue by lukejagodzinski** [microsoft/TypeScript#41684]
**TypeScript Version:**  4.1.2

**Search Terms:** `window.parent`, `window.Element`, `globalThis`

**Code**

```ts
window.HTMLDivElement;
window.parent.HTMLDivElement; // Error here
```

**Expected behavior:**

The `window.parent` object should have the same base global variables as the `window` object. If not in the context of iframe, then `window === window.parent` so all global variables should be the same in such a case. When being in the iframe context `window !== window.parent` however it should still have all the base global variables, so that we can do for example `element instanceof window.parent.Element`. You could think, why don't just write `element instanceof Element`? The answer is that it would work in almost all browsers beside Chrome. And Chrome is actually the one implementing it correctly. Both global window and iframe window have their own global context so `window.parent.Element !== window.Element`. In TS all the global variables are defined in the `globalThis` so the type for the `window.parent` should be `declare var parent: Window & typeof globalThis;`

**Actual behavior:**

The `window.parent.HTMLDivElement;` does not exist on type Window.

**Playground Link:** https://www.typescriptlang.org/play?#code/O4SwdgJg9sB0ASAVAsgGQCIgG4FEA2ApgLYFgAuA3AFCiQywAOAhgE6lkIobb7HsVA

**Comment by andrewbranch** [comment]
The problem is that then you go too far in the opposite direction; `globalThis` can easily contain things that wouldn’t be shared between an iframe and its parent. Suppose [this script](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgEITJKB1UATAewHdkIAPSEPAZ2VyuOQG8AoZd5KCOQkAGwCeyAA5wu4AFx18jAGTIwA4RAIxkAcz4EARnD4AVABbBqAbhYBfFnggI+YlADcxybRiz1CRKekzRPcgpKKmqaOnpGJuYszlDIMAQEyAC8yACMAEwAzNFufjgyRAB0ABL6ALIAMgAiwI4AonwQALYQ4OZ5HoVFouJgpRU1dY0tbWCmyAD0k8gCcAIAhCyd-t29Y0UJBEVgBADKYFCg6gAUAJQT08gEhsggBCxAA) ran in an iframe?

**Comment by lukejagodzinski** [comment]
I see your point. My solution was just a quick guess just by looking how the window object is typed. I didn't know how the `globalThis` works. My point is just that `window.parent` should have all the base global variables. Maybe the solution would be to define them in some interface and extend it by both `window` and `window.parent`. Or maybe actually put them in the `Window` interface?

**Comment by HolgerJeromin** [comment]
Same with `UIEvent.prototype.view`

**Comment by eps1lon** [comment]
> For `Document.defaultView` we had this PR which has the same problem as @andrewbranch wrote:
> [microsoft/TypeScript-DOM-lib-generator#820](https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/820) (added `& typeof globalThis`)

Yeah I see now how [this](https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/820) overshot. But `defaultView` not having `HTMLInputElement` was also annoying.
